libs/Wallet/src/SavedAddressesController.cpp

```diff
@@ -0,0 +1,58 @@
+#include "Status/Wallet/SavedAddressesController.h"
+#include "Helpers/helpers.h"
+#include "Metadata/api_response.h"
+
+#include <StatusGo/Wallet/WalletApi.h>
+
+namespace Status::Wallet
+{
+
+SavedAddressesController::SavedAddressesController(QObject* parent)
+    : QObject(parent)
+    , m_savedAddresses(Helpers::makeSharedQObject<SavedAddressesModel>("savedAddress"))
+{
+}
+
+QAbstractListModel* SavedAddressesController::savedAddresses() const
+{
+    return m_savedAddresses.get();
+}
+
+void SavedAddressesController::saveAddress(const QString &address, const QString &name)
+{
+    try
+    {
+        StatusGo::Wallet::saveAddress(StatusGo::Accounts::EOAddress(address), name);
+    }
+    catch(const StatusGo::CallPrivateRpcError& rpcError)
+    {
+        qWarning() << "StatusGoQt.saveAddress error: " <<
+            rpcError.errorResponse().error.message.c_str();
+        emit error(SaveAddressError);
+    }
+
+    // TODO: signal from wallet data_source
+    this->refresh();
+}
+
+void SavedAddressesController::refresh()
+{
+    std::vector<SavedAddressPtr> savedAddresses;
```

Comment:
why `savedAddresses` model keeps items as `shared_ptr`s? Is it necessary?

Reply:
That's common approach with QObjectVectorModel, personally i would prefer parent-child ownership